File: Programmers/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        
        List<Integer> hof = new ArrayList<>(k);
		
        for (int i = 0; i < score.length; i++) {
            hof.add(score[i]);
            
            Collections.sort(hof);
            
            if (hof.size() > k) {
                hof.remove(0);
            }
            
            answer[i] = hof.get(0);
        }
        
        return answer;
    }
}

File: Programmers/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(int k, vector<int> score) {
    vector<int> answer;
    vector<int> hof;
    
    for (int i = 0; i < score.size(); i++)
    {
        hof.push_back(score[i]);
        
        sort(hof.begin(), hof.end());
        
        if (hof.size() > k)
        {
            hof.erase(hof.begin() + 0);
        }
        
        answer.push_back(hof[0]);
    }
    
    return answer;
}